pulsar/impl_producer.go

```diff
@@ -45,14 +48,18 @@ func newProducer(client *client, options *ProducerOptions) (*producer, error) {
 		return nil, newError(ResultInvalidTopicName, "Topic name is required for producer")
 	}
 
+	if options.BatchingMaxPublishDelay == 0 {
+		options.BatchingMaxPublishDelay = defaultBatchingMaxPublishDelay
+	}
+
 	p := &producer{
 		topic: options.Topic,
 	}
 
 	if options.MessageRouter == nil {
 		internalRouter := internal.NewDefaultRouter(
 			internal.NewSystemClock(),
-			getHashingFunction(options.HashingScheme),
+			getHashingFunction(JavaStringHash),
```

Comment:
If a custom message router is not specified, we should use the default router with the configured hashing scheme.